contrib/hashicorp/vault/example_test.go

```diff
@@ -0,0 +1,95 @@
+package vault_test
+
+import (
+	"fmt"
+	"log"
+	"net/http"
+
+	"gopkg.in/DataDog/dd-trace-go.v1/contrib/hashicorp/vault"
+
+	"github.com/hashicorp/vault/api"
+)
+
+// This is the most basic way to enable tracing with Vault
+func ExampleNewHTTPClient() {
+	c, err := api.NewClient(&api.Config{
+		HttpClient: vault.NewHTTPClient(),
+		Address:    "http://vault.mydomain.com:8200",
+	})
+	if err != nil {
+		log.Fatalf("Failed to create vault client: %s\n", err)
+	}
+
+	// This call wil be traced
+	c.Logical().Read("/secret/key")
+}
+
+// Options can be passed in to configure the tracer
+func ExampleNewHTTPClient_withOptions() {
+	c, err := api.NewClient(&api.Config{
+		HttpClient: vault.NewHTTPClient(
+			vault.WithServiceName("my.vault"),
+			vault.WithAnalytics(true),
+			vault.WithAnalyticsRate(1.0)),
+		Address: "http://vault.mydomain.com:8200",
+	})
+	if err != nil {
+		log.Fatalf("Failed to create vault client: %s\n", err)
+	}
+
+	// This call wil be traced
+	c.Logical().Read("/secret/key")
+}
+
+// If you already have an *http.Client that you're using (c in this example), you can continue to use it by
+// calling WrapHTTPClient to wrap the tracing code around its Transport.
+func ExampleWrapHTTPClient() {
+	// We use a custom *http.Client to talk to vault.
+	c := &http.Client{
+		CheckRedirect: func(r *http.Request, via []*http.Request) error {
+			if len(via) > 5 {
+				return fmt.Errorf("Won't perform more that 5 redirects.")
+			}
+			return nil
+		},
+	}
+
+	client, err := api.NewClient(&api.Config{
+		HttpClient: vault.WrapHTTPClient(c),
+		Address:    "http://vault.mydomain.com:8200",
+	})
+	if err != nil {
+		log.Fatalf("Failed to create vault client: %s\n", err)
+	}
+
+	// This call wil be traced
+	client.Logical().Read("/secret/key")
+}
+
+// Options can be passed in to configure the tracer
```

Comment:
This feels a bit copy/pasted. It's better to have no documentation at all if you feel it's not needed. It's optional. I was just pointing out that you can have it in my previous PR, but most of the times it's pretty self-explanatory without it.

```go
// WrapHTTPClient can be customised further by passing it options.
```

Reply:
I've improved the sentence, and I think we should leave it in. I think it's better to be verbose, at least in the examples, since examples are there to clarify and someone using the integration might look at one example and not the other.